dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitor.java

```diff
@@ -0,0 +1,145 @@
+    {
+        private static final ESReadOnlyMonitor INSTANCE = new ESReadOnlyMonitor();
+    }
+
+    @VisibleForTesting
+    public static ESReadOnlyMonitor getInstance(
+            final SystemMessageEventUtil systemMessageEventUtil,
+            final RoleAPI roleAPI)
+    {
+        return new ESReadOnlyMonitor(systemMessageEventUtil, roleAPI);
+    }
+    public static ESReadOnlyMonitor getInstance()
+    {
+        return Singleton.INSTANCE;
+    }
+
+    public void start(final ReindexEntry reindexEntry, final String cause){
+        if (started.compareAndSet(false, true)) {
+            Logger.error(this.getClass(), Thread.currentThread().getName() + " - Reindex failed for :" + reindexEntry + " because " + cause);
+
+            if (ESIndexUtil.isAnyCurrentIndicesReadOnly()) {
+                ReindexThread.setCurrentIndexReadOnly(true);
+                sendMessage("es.index.read.only.message");
+                startMonitor();
+            } else {
+                started.set(false);
+            }
+        }
+    }
+
+    private void sendMessage(final String messageKey) {
+        try {
+            final Role adminRole = roleAPI.loadCMSAdminRole();
+            final List<String> usersId = roleAPI.findUsersForRole(adminRole)
+                    .stream()
+                    .map(user -> user.getUserId())
+                    .collect(Collectors.toList());
+
+            final SystemMessageBuilder message = new SystemMessageBuilder()
+                    .setMessage(LanguageUtil.get(messageKey))
+                    .setSeverity(MessageSeverity.ERROR)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setLife(TimeUnit.SECONDS.toMillis(5));
+
+            systemMessageEventUtil.pushMessage(message.create(), usersId);
+        } catch (final LanguageException | DotDataException | DotSecurityException e) {
+            Logger.warn(ESReadOnlyMonitor.class, () -> e.getMessage());
+        }
+    }
+
+    private void putCurrentIndicesToWriteMode() {
+        try {
+            Logger.debug(this.getClass(), "Trying to set the current indices to Write mode");
+            ESIndexUtil.putCurrentIndicesToWriteMode();
+            sendMessage("es.index.write.allow.message");
+            ReindexThread.setCurrentIndexReadOnly(false);
+
+            this.stop();
+        } catch (final ESResponseException e) {
+            Logger.info(ESReadOnlyMonitor.class, ()  -> e.getMessage());
+        }
+    }
+
+    private void startMonitor() {
+        timer = new Timer(true);
+        timer.schedule(new MonitorTimerTask(this), 0, TimeUnit.MINUTES.toMillis(1));
+    }
+
+    private void stop() {
+       this.timer.cancel();
+       this.timer = null;
+        this.started.set(false);
+    }
+
+    private static class MonitorTimerTask extends TimerTask{
+        private final ESReadOnlyMonitor esReadOnlyMonitor;
+
+        MonitorTimerTask(final ESReadOnlyMonitor esReadOnlyMonitor) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642172)